### PR TITLE
Issue 748

### LINKED ISSUE #748: Subscript/superscript
First of all, thanks a lot for this great library! It works perfectly fine in general but I found an edge case that isn't handled properly.

`canvg` renders relative font-sizes and dy values differently compared to the SVG renderer of the browsers I tested. I created a small test example:

grafik

Any idea how we can fix this?

---

This is the the problematic SVG

```
<svg id="test" width="600px" height="100px" version="1.1" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink">
  <g xmlns="http://www.w3.org/2000/svg">
    <rect fill="#67b7dc" width="600" height="100" />
    <text fill="#fff" transform="translate(20,20)" font-size="25" dy="20">
        Text and <tspan style="font-size:70%" dy="0.3em">subscript</tspan> and <tspan style="font-size:70%" dy="-0.7em">superscript</tspan>
    </text>
  </g>
</svg>
```

### PR DESCRIPTION
**Pull request by gabelerner**
fixes #748

### CODE REVIEW

File: src/Property.ts [hunk 1]
```diff
@@ -147,7 +147,7 @@ export default class Property<T = any> {
 		return this.getString().replace(/[0-9\.\-]/g, '');
 	}
 
-	getPixels(axis?: Axis, processPercent = false) {
+	getPixels(axis?: Axis, processPercent = false, considerParentFontSize = false) {
```
Comment (gabelerner) [inline src/Property.ts]: @dangreen i took this from https://github.com/canvg/canvg/pull/749 ... we can also choose to use named options here to not have to pass in double `undefined`.

Comment (dangreen) [reply src/Property.ts]: @gabelerner 

```ts
getPixels(axis?: Axis, processPercent?: boolean);
getPixels(isFontSize?: boolean);
getPixels(axisOrIsFontSize?: Axis | boolean, processPercent = false) {

  const [
    axis,
    isFontSize
  ] = typeof axisOrIsFontSize === 'boolean'
    ? [undefined, axisOrIsFontSize]
    : [axisOrIsFontSize];
  ...
    case this.isString(/%$/) && isFontSize:
      return this.getNumber() * this.getEm(/* viewPort */);
   
    case this.isString(/%$/):
      this.getNumber() * viewPort.computeSize(axis);
  ...
}
```

File: src/Property.ts [hunk 2]
```diff
@@ -147,7 +147,16 @@ export default class Property<T = any> {
 		return this.getString().replace(/[0-9\.\-]/g, '');
 	}
 
-	getPixels(axis?: Axis, processPercent = false) {
+	getPixels(axis?: Axis, processPercent?: boolean): number;
+	getPixels(isFontSize?: boolean): number;
+	getPixels(axisOrIsFontSize?: Axis | boolean, processPercent = false): number {
```
Comment (gabelerner) [inline src/Property.ts]: @dangreen linter complained had to add `: number` return type 🤷‍♂

File: src/Property.ts [hunk 3]
```diff
@@ -212,6 +221,9 @@ export default class Property<T = any> {
 			case this.isString(/in$/):
 				return this.getNumber() * this.getDpi(/* viewPort */);
 
+			case this.isString(/%$/) && isFontSize:
```
Comment (gabelerner) [inline src/Property.ts]: diff looks prettier but now we're running the `%` regex twice.

File: src/Property.ts (hunk 2 above)
Comment (dangreen) [reply src/Property.ts]: It's ok, my mistake

File: src/Property.ts (hunk 3 above)
Comment (dangreen) [reply src/Property.ts]: Do it like you want, by `case` or `if` 😃

Comment (gabelerner) [reply src/Property.ts]: i'll keep style ... we probably want to use `endsWith` or something in the long run if we want higher performance here over regex
